samples/collect_data/collect_data.cpp

```diff
@@ -0,0 +1,196 @@
+            return conn;
+        }
+    }
+    return NULL;
+}
+
+bool is_connected(sockpp::connector* array_connection[], int arrSize) {
+    if (NULL == array_connection || 0 == arrSize) {
+        return NULL;
+    }
+    for (int i = 0; i < arrSize; ++i) {
+        sockpp::connector* conn = array_connection[i];
+        if (conn != NULL && conn->is_connected()) {
+            return true;
+        }
+    }
+    return false;
+}
+
+int main(int, char**) {
+    //boinc_init();
+
+    sockpp::socket_initializer sockInit;
+    sockpp::tcp_connector tcp_conn;
+    sockpp::connector* array_connection[] = { &tcp_conn, NULL };
+#if !_WIN32
+    sockpp::unix_connector unix_conn;
+    array_connection[1] = &unix_conn;
+#endif
+    sockpp::connector* conn = NULL;
+    const int arrSize = sizeof(array_connection) / sizeof(array_connection[0]);
+    const bool isDebug = false;
+    const int16_t port = 31416;
+    const int buffer_size = 4096;
+    const int wake_minutes = 30;
+    const int exec_hourly = 5;
+    int suffix_file_idx = 0;
+    const std::string android_address = "edu_berkeley_boinc_client_socket";
+    const std::string unix_address = "boinc_socket";
+    const std::string tcp_address = "localhost";
+    const std::string prefix_file_name = "boinc";
+    const std::chrono::minutes wake_every_minutes = std::chrono::minutes(wake_minutes);
+    const std::chrono::minutes exec_time_hours = std::chrono::minutes(exec_hourly);
+    
+    std::stringstream stream;
+    char buffer[buffer_size];
+    const std::chrono::time_point<std::chrono::system_clock> starttime = std::chrono::system_clock::now();
+    while (true) {
+        int attemped_connections = 0;
+        while (!is_connected(array_connection, arrSize)) {
+            attemped_connections++;
+            std::cout << "attemped connections: " << attemped_connections << "\n";
+            if (!tcp_conn.connect(sockpp::inet_address(tcp_address, port))) {
+                std::cout << "tcp connection fail:\n";
+                std::cout << tcp_conn.last_error_str() << "\n";
+                tcp_conn.clear();
+            }
+#if! _WIN32
+            if (!is_connected(array_connection, arrSize) && !unix_conn.connect(sockpp::unix_address(unix_address))) {
+                std::cout << "unix connection fail:\n";
+                std::cout << unix_conn.last_error_str() << "\n";
+                unix_conn.clear();                
+            }
+
+            if (!is_connected(array_connection, arrSize) && !unix_conn.connect(sockpp::unix_address(android_address))) {
+                std::cout << "android unix connection fail:\n";
+                std::cout << unix_conn.last_error_str() << "\n";
+                unix_conn.clear();                
+            }
+#endif
+            if (is_connected(array_connection, arrSize)) {
+                conn = get_connected(array_connection, arrSize);
+            }
+            std::cout << "\n";
+            my_sleep(std::chrono::seconds(1));
```

Comment:
Don't you think that doing this every second is necessary?

Reply:
Yes.